webapp/tests/karma/ts/modules/analytics/analytics.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { ActivatedRoute, Router } from '@angular/router';
+import { provideMockStore } from '@ngrx/store/testing';
+import sinon from 'sinon';
+import { expect } from "chai";
+
+import { AnalyticsComponent } from '@mm-modules/analytics/analytics.component';
+import { AnalyticsModulesService } from '@mm-services/analytics-modules.service';
+import { GlobalActions } from '@mm-actions/global';
+import { AnalyticsActions } from '@mm-actions/analytics';
+
+describe('AnalyticsComponent', () => {
+  let component: AnalyticsComponent;
+  let fixture: ComponentFixture<AnalyticsComponent>;
+  let analyticsModulesService;
+  let globalActions;
+  let analyticsActions;
+  let activatedRoute;
+  let router;
+
+  beforeEach(async(() => {
+    analyticsModulesService = { get: sinon.stub().resolves([]) };
+    const mockSelectors = [
+      { selector: 'getAnalyticsModules', value: [] },
+    ];
+    globalActions = {
+      unsetSelected: sinon.stub(GlobalActions.prototype, 'unsetSelected')
+    };
+    analyticsActions = {
+      setSelectedAnalytics: sinon.stub(AnalyticsActions.prototype, 'setSelectedAnalytics'),
+      setAnalyticsModules: sinon.stub(AnalyticsActions.prototype, 'setAnalyticsModules')
+    };
+    activatedRoute = {
+      snapshot: {
+        routeConfig: { path: 'targets' },
+      },
+    };
+
+    TestBed.configureTestingModule({
+      declarations: [ AnalyticsComponent ],
+      imports: [
+        RouterTestingModule,
+      ],
+      providers: [
+        provideMockStore({ selectors: mockSelectors }),
+        { provide: AnalyticsModulesService, useValue: analyticsModulesService },
+        { provide: ActivatedRoute, useValue: activatedRoute },
+      ]
+    })
+    .compileComponents()
+    .then(() => {
+      fixture = TestBed.createComponent(AnalyticsComponent);
+      component = fixture.componentInstance;
+      analyticsModulesService = TestBed.inject(AnalyticsModulesService);
```

Comment:
I don't think this is needed here. You've initialized the value up top.

Reply:
Right, fixed!